tortoise/algorithm.go

```diff
@@ -120,6 +120,11 @@ func (t *Tortoise) OnBeacon(eid types.EpochID, beacon types.Beacon) {
 	t.mu.Lock()
 	defer t.mu.Unlock()
 	evicted := t.trtl.evicted.GetEpoch()
+	// when the node recover from checkpoint, the genesis epoch and recover epoch can be the same
+	if types.GetEffectiveGenesis() != types.FirstEffectiveGenesis() &&
+		evicted == (types.GetEffectiveGenesis()+1).GetEpoch() {
+		evicted--
```

Comment:
note that `evicted` here is a local variable for epoch, i am not changing `t.trtl.evicted`.

took a closer look at the eviction logic. it seems that the current code at develop isn't consistent.
say epoch size is 10. window size is 100. ideally data for epoch 2 should be removed when t.trtl.evicted == 29.
but the current code will disregard `OnBeacon()` for epoch 2 when t.trtl.evicted == 20.

i think the code should simply change to 
```go
evicted := t.trtl.evicted.Add(1).Epoch()  // we still care about t.trtl.evicted.Add(1)
if eid < evicted {  // note "<" instead of "<<"
  return
} 
```
this will also work for the recovery case.
the calculation in `lookbackWindowStart()` needs to be adjusted for recovery tho.

Reply:
> evicted := t.trtl.evicted.Add(1).Epoch()  // we still care about t.trtl.evicted.Add(1)
if eid < evicted {  // note "<" instead of "<<"
  return
} 

yes, this looks good to me as well